### ISSUE TITLE
Add Java 17 support

### ISSUE THREAD

**Issue by mrjameshamilton** [Guardsquare/proguard#132]

**Comment by dougxc** [comment]
Hi @mrjameshamilton , is this currently under development using the JDK 17 EA binaries? Is there a rough ETA as to how soon after the JDK 17 release (in mid Sept) this will be resolved?

**Comment by dougxc** [comment]
Awesome to see the [7.2.0-beta 1 release](https://github.com/Guardsquare/proguard/releases/tag/v7.2.0-beta1).
Are there plans to make it available via maven?

**Comment by mrjameshamilton** [comment]
Hi @dougxc ! It should already be available via Maven. I noticed a problem when executing ProGuard on Java 9+ though (`java.lang.UnsupportedOperationException: No class provided, and an appropriate one cannot be found`), so we might have to make a beta2 soon to fix that.

**Comment by dougxc** [comment]
Am I looking the wrong place? https://mvnrepository.com/artifact/com.guardsquare/proguard-base

**Comment by mrjameshamilton** [comment]
I guess it hasn't yet synced. The artefacts are there though e.g. https://repo.maven.apache.org/maven2/com/guardsquare/proguard-base/7.2.0-beta1/proguard-base-7.2.0-beta1.jar

**Comment by dougxc** [comment]
The `7.2.0-beta1` artefacts for `proguard-base` and `proguard-retrace` are available. However, for [`proguard-core`](https://repo.maven.apache.org/maven2/com/guardsquare/proguard-core/), I see a `8.0.0` artefact but nothing for `7.2.0-beta1`.

**Comment by mrjameshamilton** [comment]
Indeed, ProGuard 7.2.0-beta1 has a dependency on ProGuardCORE 8.0.0; we've unlinked the versioning of the two projects, since ProGuardCORE is a general library that is used for other things as well as ProGuard; and there were some breaking changes in ProGuardCORE for which it seemed appropriate to do a major version bump.
